File: src/macos/AzPin/Utilities/ResourceTypeMapper.swift
import Foundation

enum ResourceTypeMapper {
    static let runnableTypes: Set<String> = [
        "microsoft.web/sites",
        "microsoft.web/sites/slots",
        "microsoft.app/containerapps",
        "microsoft.logic/workflows"
    ]

    static func symbolName(for resourceType: String) -> String {
        switch resourceType.lowercased() {
        case "microsoft.web/sites":                        return "globe"
        case "microsoft.insights/components":              return "lightbulb.fill"
        case "microsoft.storage/storageaccounts":          return "externaldrive.fill"
        case "microsoft.servicebus/namespaces":            return "arrow.triangle.branch"
        case "microsoft.keyvault/vaults":                  return "key.fill"
        case "microsoft.apimanagement/service":            return "antenna.radiowaves.left.and.right"
        case "microsoft.sql/servers",
             "microsoft.documentdb/databaseaccounts":      return "cylinder.fill"
        case "microsoft.app/containerapps":                return "shippingbox.fill"
        case "microsoft.web/serverfarms":                  return "bolt.fill"
        default:                                           return "cloud.fill"
        }
    }

    static func isRunnable(_ resourceType: String) -> Bool {
        runnableTypes.contains(resourceType.lowercased())
    }
}

File: src/windows/AzPin.Windows.Core/Utilities/ResourceTypeMapper.cs
namespace AzPin.Windows.Utilities;

public static class ResourceTypeMapper
{
    private static readonly Dictionary<string, string> TypeToGlyph =
        new(StringComparer.OrdinalIgnoreCase)
    {
        ["microsoft.resources/resourcegroups"]   = "", // Folder
        ["microsoft.web/sites"]                  = "", // Globe
        ["microsoft.web/sites/slots"]            = "",
        ["microsoft.insights/components"]        = "", // Lightbulb
        ["microsoft.storage/storageaccounts"]    = "", // Storage
        ["microsoft.servicebus/namespaces"]      = "", // Branch
        ["microsoft.keyvault/vaults"]            = "", // Key
        ["microsoft.apimanagement/service"]      = "", // Antenna
        ["microsoft.sql/servers"]                = "", // Database
        ["microsoft.documentdb/databaseaccounts"] = "",
        ["microsoft.app/containerapps"]          = "", // Box/Package
        ["microsoft.logic/workflows"]            = "", // Flow
    };

    private static readonly HashSet<string> RunnableTypes = new(StringComparer.OrdinalIgnoreCase)
    {
        "microsoft.web/sites",
        "microsoft.web/sites/slots",
        "microsoft.app/containerapps",
        "microsoft.logic/workflows"
    };

    private const string DefaultGlyph = ""; // Cloud

    public static string GlyphFor(string resourceType) =>
        TypeToGlyph.GetValueOrDefault(resourceType.ToLowerInvariant(), DefaultGlyph);

    public static bool IsRunnable(string resourceType) =>
        RunnableTypes.Contains(resourceType);
}
